test/test_pauli.jl

```diff
@@ -5,11 +5,10 @@ using QuantumOpticsBase
 
 @testset "pauli" begin
 
-@test_throws MethodError PauliBasis(1.4)
-
+b = SpinBasis(1//2)
 # Test conversion of unitary matrices to superoperators.
-q2 = PauliBasis(2)
-q3 = PauliBasis(3)
+q2 = b^2
+q3 = b^3
```

Comment:
`PauliBasis` is just being renamed, not completely removed, right? There is still value to having a convenient way to represent multi-qubit basis. This probably should just use the new name. Same for the other changes.